### PR TITLE
Implemented option for serializer max_depth

### PR DESCRIPTION
**Pull request by dle-sweethearts**
Adds the possibility to define the default depth of the array serialization via configuration

### CODE REVIEW

**Comment by nokitakaze** [comment]
Don't you want to fix code style issues?

File: lib/Raven/Serializer.php [hunk 1]
```diff
@@ -39,20 +39,37 @@ class Raven_Serializer
...
      * @var string
      */
     protected $mb_detect_order = self::DEFAULT_MB_DETECT_ORDER;
 
     /**
+     * The maximum depth when serializing arrays/objects
+     *
+     * @var
+     */
+    protected $max_depth = self::DEFAULT_MAX_DEPTH;
```
Comment (nokitakaze) [inline lib/Raven/Serializer.php]: Protected properties should begin with underscore

File: lib/Raven/Serializer.php [hunk 2]
```diff
@@ -39,20 +39,37 @@ class Raven_Serializer
...
      * @param null|string $mb_detect_order
+     * @param null|int    $max_depth
      */
-    public function __construct($mb_detect_order = null)
+    public function __construct($mb_detect_order = null, $max_depth = null )
     {
         if ($mb_detect_order != null) {
             $this->mb_detect_order = $mb_detect_order;
         }
+        if ( null !== $max_depth ) {
```
Comment (nokitakaze) [inline lib/Raven/Serializer.php]: maybe !is_null($max_depth) ?

File: lib/Raven/Serializer.php [hunk 3]
```diff
@@ -64,8 +81,9 @@ public function __construct($mb_detect_order = null)
      * @param int   $_depth
      * @return string|bool|double|int|null|object|array
      */
-    public function serialize($value, $max_depth = 3, $_depth = 0)
+    public function serialize($value, $max_depth = null, $_depth = 0)
     {
+        $max_depth = null === $max_depth ? $this->max_depth : $max_depth;
```
Comment (nokitakaze) [inline lib/Raven/Serializer.php]: Maybe
```php
$max_depth = !is_null($max_depth) ? $max_depth : $this->max_depth;
```

?

**Comment by nokitakaze** [comment]
Okay. I need to add this pr to #454

**Comment by stayallive** [comment]
This was done a long time ago and never updated with the latest changes so we cannot move forward with this PR right now. If it's updated we can reconsider it. Sorry it got left in the cold.

**Pull request closed without merging**